airflow/www/security.py

```diff
@@ -18,16 +18,23 @@
 #
 
 import warnings
+from datetime import timedelta
 from typing import Dict, Optional, Sequence, Set, Tuple
 
-from flask import current_app, g
+import jwt
+from flask import current_app, g, request
+from flask_appbuilder.const import AUTH_DB, AUTH_LDAP, AUTH_OAUTH, AUTH_REMOTE_USER
 from flask_appbuilder.security.sqla import models as sqla_models
 from flask_appbuilder.security.sqla.manager import SecurityManager
 from flask_appbuilder.security.sqla.models import PermissionView, Role, User
+from flask_jwt_extended import JWTManager, create_access_token, create_refresh_token
+from flask_login import login_user
 from sqlalchemy import or_
 from sqlalchemy.orm import joinedload
 
 from airflow import models
+from airflow.api_connexion.exceptions import Unauthenticated
+from airflow.api_connexion.schemas.auth_schema import auth_schema
```

Comment:
Hmmmm, having this import code from api_connextion module is a bit of a "code smell".

@jedcunningham @jhtimmins Do you have any thoughts here? I know you were discussing somewhere or other if we needed our own Airflow security layer separate from FAB.

Reply:
I tend to agree. I think a slight refactor could keep the security manager cleaner and move the api_connextion specific stuff back to that side.

e.g, take a look at how [auth_user_ldap](https://github.com/dpgaspar/Flask-AppBuilder/blob/27b15e59316e85e0fe62b8aa9978391ed4c729c9/flask_appbuilder/security/manager.py#L990) just returns `None` and lets the caller deal with the right response.

Reply:
Agree. This was why I kept them at api_connexion/endpoints/ until this commit 724c90939e3be8dccaaea9a5c92582397d26ac81

Moving it will mean that some other codes would be moved. That's actually how flask appbuilder handled the views  https://github.com/dpgaspar/Flask-AppBuilder/blob/27b15e59316e85e0fe62b8aa9978391ed4c729c9/flask_appbuilder/security/views.py#L624

Happy to move it back if you accept @ashb

Reply:
> Agree. This was why I kept them at api_connexion/endpoints/ until this commit [724c909](https://github.com/apache/airflow/commit/724c90939e3be8dccaaea9a5c92582397d26ac81)
> 
> Moving it will mean that some other codes would be moved. That's actually how flask appbuilder handled the views https://github.com/dpgaspar/Flask-AppBuilder/blob/27b15e59316e85e0fe62b8aa9978391ed4c729c9/flask_appbuilder/security/views.py#L624
> 
> Happy to move it back if you accept @ashb

Taking back this and working on a way around it now

Reply:
Yeah, sounds good.

Sometimes I have to see the code to realise it's wrong, sorry for mixed signals :)